packages/next/src/server/app-render/create-component-tree.tsx

```diff
@@ -636,23 +706,58 @@ async function createComponentTreeInternal({
               </>
             ),
           }
-          const notfoundClientSegment = (
-            <ClientSegmentRoot
-              Component={SegmentComponent}
-              slots={notFoundParallelRouteProps}
-              params={currentParams}
-            />
+          notfoundClientSegment = (
+            <>
+              {layerAssets}
+              <ClientSegmentRoot
+                Component={SegmentComponent}
+                slots={notFoundParallelRouteProps}
+                params={currentParams}
+              />
+            </>
           )
-
+        }
+        if (Forbidden) {
+          const forbiddenParallelRouteProps = {
+            children: forbiddenElement,
+          }
+          forbiddenClientSegment = (
+            <>
+              {layerAssets}
+              <ClientSegmentRoot
+                Component={SegmentComponent}
+                slots={forbiddenParallelRouteProps}
+                params={currentParams}
+              />
+            </>
+          )
+        }
+        if (Unauthorized) {
+          const unauthorizedParallelRouteProps = {
+            children: unauthorizedElement,
+          }
+          unauthorizedClientSegment = (
+            <>
+              {layerAssets}
+              <ClientSegmentRoot
+                Component={SegmentComponent}
+                slots={unauthorizedParallelRouteProps}
+                params={currentParams}
+              />
+            </>
+          )
+        }
```

Comment:
yea it's the same issue like always having the server component not-found payload in the first html, and @gnoff was suggesting sth like serving those rsc payload through another endpoints, so they can be downloaded later. I feel that could be addressed as a separate work to deal with it

Reply:
sounds good, yeah I we don't have to do that here, just flagging that I think it should block stable since we already have a problem in the notFound case and this'll make it worse